### PR TITLE
Fix incorrect conversion result caused by BOM

### PR DESCRIPTION
**Pull request by moria97**
The conversion result could be wrong when user save template files with UTF8 with BOM encoding because the compress procedure does not well handle the BOM characters.
Add test cases to make sure conversion results from template folder and template image are the same.

### CODE REVIEW

**Comment by ACMoretxj** [comment]
A little weird that my local env works perfectly for testing while the pipeline complains with `Microsoft.Health.Fhir.Liquid.Converter.Exceptions.ConverterInitializeException : Template folder not found: ..\..\..\..\..\data\Templates\Hl7v2.`.

File: src/Microsoft.Health.Fhir.TemplateManagement.FunctionalTests/TemplateCollectionFunctionalTests.cs [hunk 1]
```diff
@@ -264,6 +273,25 @@ public async Task GiveDefaultImageReference_WhenGetTemplateCollectionWithEmptyTo
...
+            var templateProviderFactory = new TemplateCollectionProviderFactory(new MemoryCache(new MemoryCacheOptions()), Options.Create(new TemplateCollectionConfiguration()));
+            var templateProvider = templateProviderFactory.CreateTemplateCollectionProvider(ImageInfo.DefaultTemplateImageReference, string.Empty);
+            var imageTemplateProvider = new Hl7v2TemplateProvider(await templateProvider.GetTemplateCollectionAsync(CancellationToken.None));
+
+            var hl7v2Processor = new Hl7v2Processor();
+            var inputContent = File.ReadAllText(inputFile);
+
+            var imageResult = hl7v2Processor.Convert(inputContent, rootTemplate, imageTemplateProvider);
+            var folderResult = hl7v2Processor.Convert(inputContent, rootTemplate, folderTemplateProvider);
+            Assert.Equal(imageResult, folderResult);
```
Comment (ACMoretxj) [inline src/Microsoft.Health.Fhir.TemplateManagement.FunctionalTests/TemplateCollectionFunctionalTests.cs:292]: These two strings are JSON objects actually, can we compare them by string.comparison?

**Comment by sowu880** [comment]
> A little weird that my local env works perfectly for testing while the pipeline complains with `Microsoft.Health.Fhir.Liquid.Converter.Exceptions.ConverterInitializeException : Template folder not found: ..\..\..\..\..\data\Templates\Hl7v2.`.

Will be fixed in PR #172

**Comment by ACMoretxj** [comment]
> > A little weird that my local env works perfectly for testing while the pipeline complains with `Microsoft.Health.Fhir.Liquid.Converter.Exceptions.ConverterInitializeException : Template folder not found: ..\..\..\..\..\data\Templates\Hl7v2.`.
> 
> Will be fixed in PR #172

Got it, thx.

File: src/Microsoft.Health.Fhir.TemplateManagement.FunctionalTests/TemplateCollectionFunctionalTests.cs (hunk 1 above)
Comment (ACMoretxj) [reply src/Microsoft.Health.Fhir.TemplateManagement.FunctionalTests/TemplateCollectionFunctionalTests.cs:303]: Update: Currently we are using a lot of this kind of comparison methods, but only in testing. I think it's ok.

**Review by ACMoretxj — APPROVED** [review]

File: src/Microsoft.Health.Fhir.TemplateManagement.FunctionalTests/TemplateCollectionFunctionalTests.cs (hunk 1 above)
Comment (moria97) [reply src/Microsoft.Health.Fhir.TemplateManagement.FunctionalTests/TemplateCollectionFunctionalTests.cs:303]: Is it possible that same inputdata will be serialized to different json string? I feel the serialization result should be stable

**Review by BoyaWu10 — APPROVED** [review]

**Pull request merged**